cep-recipe-serialization.md

```diff
@@ -0,0 +1,1110 @@
+
+### Build configuration section
+
+This records all information that was used at build time, incl. variables given
+on the CLI or by environment yaml. Everything necessary to reproduce the build
+should be recorded here.
+
+Here's the documentation for the fields in the provided YAML:
+
+#### `build_configuration`
+
+The `build_configuration` section contains information related to the build
+process.
+
+##### `target_platform`
+
+- Type: string
+- Description: The target platform for which the package is being built (e.g.,
+  `osx-arm64`).
+
+##### `host_platform`
+
+- Type: string
+- Description: The host platform used for building the package. Usually
+  equivalent to `target_platform`, except when the `target_platform` is
+  `noarch`. If the `target_platform` is `noarch`, the `host_platform` is the
+  platform where the build is being executed.
+
+##### `build_platform`
+
+- Type: string
+- Description: The platform used for building the package (e.g., `osx-arm64`).
+
+##### `variant`
+
+- Type: object
+- Description: Dictionary with the variant configuration used for the build.
+
+##### `hash`
+
+- Type: object
+- Description: Contains information related to hashing.
+  - `hash`: The hash value (e.g., `60d57d3`).
+  - `hash_input`: The input values used for generating the hash.
+  - `hash_prefix`: Prefix for the hash, which can include values like `py311`
+    based on the variant configuration.
+
+##### `directories`
+
+- Type: object
+- Description: Contains directory paths used during the build process.
+  - `host_prefix`: The host prefix directory with placeholders for the host
+    environment.
+  - `build_prefix`: The build prefix directory for the build environment.
+  - `work_dir`: The working directory with extracted sources.
+  - `build_dir`: The build directory that contains `host_prefix`,
+    `build_prefix`, and `work_dir` as subdirectories.
+
+##### `channels`
+
+- Type: array of strings
+- Description: The conda channels used for resolving dependencies (e.g.,
+  `conda-forge`).
+
+##### `timestamp`
+
+- Type: string / timestamp in ISO 8601 format
+- Description: The timestamp of the build configuration (e.g.,
+  `2024-04-13T14:35:30.774863Z`).
+
+##### `subpackages`
+
+- Type: list of objects
+- Description: Contains information about subpackages that are used in
+  `pin_subpackage` expressions.
+  - `curl`: An example subpackage.
+    - `name`: The name of the subpackage (e.g., `curl`).
+    - `version`: The version of the subpackage (e.g., `8.0.1`).
+    - `build_string`: The build string for the subpackage (e.g., `h60d57d3_0`).
+
+##### `packaging_settings`
+
+- Type: object
+- Description: Contains settings related to the final package output.
+  - `archive_type`: The type of archive used for packaging (either `tar_bz2` or
+    `conda`).
+  - `compression_level`: The numerical compression level used for packaging
+    (e.g., `15`).
+  - `compression_threads`: The number of threads used for compression (e.g.,
+    `1`).
```

Comment:
Is there a reason to preserve the number of compression threads? I honestly would not expect whatever compression algorithm we use to change outputs based on the number of threads used.

Reply:
yeah it was more about being overly cautious. We can run some tests wether the number of threads has any effect.

Reply:
We ended up running these tests, as claimed by zstd there was no difference so we removed this key.